Repeating branching logic

ihabzee opened the pull request:
Hi @mmcev106, @grezniczek, 

I implemented a fix to EM branching logic on repeating instances. Which will allow branching logic condition to be executed only on the correct level based on the branching logic field level. For example from below json, If you type USA as a country then states repeating instance fields will show up. And if you add another country instance states list wont show up. Same behavior will happen if you select CA then descriptive field will show up only within CA boundaries. 

I am available to go over this if you have any question or need more details.  

Test EM config.json below: 

`	"project-settings": [
      {
        "key": "top-level",
        "name": "Top Level",
        "type": "checkbox"
      },
      {
        "key": "child-project-id",
        "name": "Projects List",
        "required": true,
        "type": "project-id"
      },
      {
        "key": "countries",
        "name": "Country",
        "type": "sub_settings",
        "repeatable": true,
        "sub_settings": [
          {
            "key": "country-name",
            "name": "Country Name (on the bottom just to test a different order)",
            "type": "text"
          },
          {
            "key": "states",
            "name": "State",
            "type": "sub_settings",
            "repeatable": true,
            "branchingLogic": {
              "field": "country-name",
              "op": "=",
              "value": "USA"
            },
            "sub_settings": [
              {
                "key": "state-name",
                "name": "State Name",
                "type": "dropdown",
                "choices": [
                  {
                    "name": "Arizona",
                    "value": "AZ"
                  },
                  {
                    "name": "California",
                    "value": "CA"
                  }
                ]
              },
              {
                "key": "state-top-level",
                "name": "Top Level is True",
                "type": "descriptive",
                "branchingLogic": {
                  "field": "top-level",
                  "op": "=",
                  "value": "1"
                }
              },
              {
                "key": "cities",
                "name": "City",
                "type": "sub_settings",
                "repeatable": true,
                "sub_settings": [
                  {
                    "key": "city-name",
                    "name": "City Name",
                    "type": "text"
                  },
                  {
                    "key": "desc-california",
                    "name": "You are in California",
                    "type": "text",
                    "branchingLogic": {
                      "field": "state-name",
                      "op": "=",
                      "value": "CA"
                    }
                  },
                  {
                    "key": "desc-usa",
                    "name": "You are in the USA",
                    "type": "descriptive",
                    "branchingLogic": {
                      "field": "country-name",
                      "op": "=",
                      "value": "USA"
                    }
                  },
                  {
                    "key": "city-size",
                    "name": "City Size",
                    "type": "text"
                  }
                ]
              }
            ]
          }
        ]
      }
	],
`

Thanks, 

@123andy

mmcev106 commented:
Thanks very much @ihabzee!  Setting branching logic definitely needed some love.  It looks like there are some conflicts with the `testing` branch preventing merging unfortunately.  Would you mind merging the latest version of `testing` into your branch?  It looks like your editor might have normalized a bunch of whitespace, which might make a lot more merge conflicts than necessary.  I'm happy merging whitespace changes, but just wanted to let you know in case you wanted to redo this without the whitespace changes just to avoid the conflicts.

ihabzee commented:
@mmcev106 I updated my pull request with `testing` branch. Regarding white spaces I added .editorconfig to try to mimc your coding style but PHPStorm adds more stuff when I try to enforce the coding styles. So would you mind test and merge existing code? 

Thanks,

mmcev106 commented:
I would be very quick to merge a PR that is only about fixing branching logic.  However, it looks like this is mostly about introducing/enforcing a coding standard.  I do not feel comfortable taking the time to review, test, & merge this without my team's approval of this standard, and maybe some sort of automated way to enforce it.  I will bring it up in our team meeting on Monday since it's time to revisit this anyway.  However, a discussion could take weeks, and even then we may decided a coding standard is not worth the effort to enforce.  If we do approve a standard, there will likely be some differences from what the PR currently contains (for example, we may follow [StandardJS](https://standardjs.com/) which would exclude the semicolons this PR adds currently).

If you would like the branching logic changes to get merged sometime soon, they will likely need to be split out into their own PR separate from all the other changes.
